### PR TITLE
Cleanup and document `-C code-model`

### PR DESCRIPTION
**Pull request by petrochenkov**
r? @Amanieu

### CODE REVIEW

File: src/doc/rustc/src/codegen-options/index.md [hunk 1]
```diff
@@ -9,9 +9,25 @@ This option is deprecated and does nothing.
...
 
-To find the valid options for this flag, run `rustc --print code-models`.
+The specific ranges depend on target architectures and addressing modes available to them. \
+For x86 more detailed description of its code models can be found in
+[System V Application Binary Interface](https://github.com/hjl-tools/x86-psABI/wiki/x86-64-psABI-1.0.pdf)
+specification.
+
+Supported values for this option are:
+
+<!-- - `tiny` - Tiny code model. -->
```
Comment (Amanieu) [inline src/doc/rustc/src/codegen-options/index.md:24]: Is there a reason why we don't expose the tiny code model to users? AFAIK it's only available on AArch64 where it assumes all symbols are within 1MB of each other.

**Review by Amanieu — APPROVED** [review]
r=me

File: src/doc/rustc/src/codegen-options/index.md (hunk 1 above)
Comment (petrochenkov) [reply src/doc/rustc/src/codegen-options/index.md:24]: As I understand, it was simply introduced to LLVM later than the `code-model` option had been implemented in rustc.

I'll make a separate PR exposing it, I didn't want to do it in a "cleanup and document" PR.

**Pull request merged**